lib/iris/tests/unit/fileformats/rules/test__make_cube.py

```diff
@@ -57,9 +61,9 @@ def test_invalid_units(self):
         self.assertEqual(cube.attributes, expected_attributes)
 
         # Check warning was raised.
-        self.assertEqual(warn.call_count, 1)
-        warning_msg = warn.call_args[0][0]
-        self.assertIn('invalid units', warning_msg)
+        self.assertEqual(len(warn), 1)
+        exp_emsg = 'invalid units {!r}'.format(units)
+        six.assertRegex(self, str(warn[0].message), exp_emsg)
```

Comment:
We could, but there are a [couple of](https://github.com/SciTools/iris/blob/dask/lib/iris/tests/unit/fileformats/grib/save_rules/test_set_time_range.py#L100) [other examples](https://github.com/SciTools/iris/blob/dask/lib/iris/tests/unit/fileformats/pp/test__field_gen.py#L76) of this test method being used when testing warning text like this (including the one I duplicated here). Admittedly there are actually more examples of `assertRegexpMatches` being used in such situations, so I'll change this.